src/libLookup/Lookup.cpp

```diff
@@ -1623,8 +1625,15 @@ bool Lookup::ProcessSetDSBlockFromSeed(const bytes& message,
     }
 
     for (const auto& dsblock : dsBlocks) {
-      if (!(m_mediator.m_dsBlockChain.GetBlock(
-                dsblock.GetHeader().GetBlockNum()) == DSBlock())) {
+      // TODO
+      // Workaround to identify dummy block as == comparator does not work on
+      // empty object for DSBlock and DSBlockheader().
+      // if (!(m_mediator.m_dsBlockChain.GetBlock(
+      //           dsblock.GetHeader().GetBlockNum()) == DSBlock())) {
+      if (!(m_mediator.m_dsBlockChain
+                .GetBlock(dsblock.GetHeader().GetBlockNum())
+                .GetHeader()
+                .GetBlockNum() == INIT_BLOCK_NUMBER)) {
```

Comment:
Change this to (m_mediator.m_dsBlockChain
                .GetBlock(dsblock.GetHeader().GetBlockNum())
                .GetHeader()
                .GetBlockNum() != INIT_BLOCK_NUMBER